src/awkward/_v2/contents/numpyarray.py

```diff
@@ -395,6 +395,300 @@ def iscontiguous(self):
 
         return True
 
+    def _subranges_equal(self, starts, stops, length, sorted=True):
+        nplike = self.nplike
+
+        is_equal = ak._v2.index.Index64.zeros(1, nplike)
+
+        tmp = ak._v2.contents.NumpyArray(nplike.empty(length, self.dtype))
+        self._handle_error(
+            nplike[
+                "awkward_NumpyArray_fill",
+                tmp._data.dtype.type,
+                self._data.dtype.type,
+            ](
+                tmp._data,
+                0,
+                self._data,
+                length,
+            )
+        )
+
+        if not sorted:
+            tmp_beg_ptr = ak._v2.index.Index64.empty(ak._util.kMaxLevels, nplike)
+            tmp_end_ptr = ak._v2.index.Index64.empty(ak._util.kMaxLevels, nplike)
+
+            self._handle_error(
+                nplike[
+                    "awkward_quick_sort",
+                    tmp._data.dtype.type,
+                    tmp_beg_ptr.dtype.type,
+                    tmp_end_ptr.dtype.type,
+                    starts.dtype.type,
+                    stops.dtype.type,
+                ](
+                    tmp._data,
+                    tmp_beg_ptr.to(nplike),
+                    tmp_end_ptr.to(nplike),
+                    starts.to(nplike),
+                    stops.to(nplike),
+                    True,
+                    len(starts),
+                    ak._util.kMaxLevels,
+                )
+            )
+        self._handle_error(
+            nplike[
+                "awkward_NumpyArray_subrange_equal",
+                tmp._data.dtype.type,
+                starts.dtype.type,
+                stops.dtype.type,
+                np.bool_,
+            ](
+                tmp._data,
+                starts.to(nplike),
+                stops.to(nplike),
+                len(starts),
+                is_equal.to(nplike),
+            )
+        )
+
+        return True if is_equal[0] == 1 else False
+
+    def _as_unique_subranges(self, offsets, length):
+        nplike = self.nplike
+
+        nextoffsets = ak._v2.index.Index64.zeros(length, nplike)
+        self._handle_error(
+            nplike[
+                "awkward_unique_ranges",
+                self._data.dtype.type,
+                offsets.dtype.type,
+                nextoffsets.dtype.type,
+            ](
+                self._data,
+                len(self._data),
+                offsets.to(nplike),
+                length,
+                nextoffsets.to(nplike),
+            )
+        )
+
+        out2 = self[0 : nextoffsets[-1]]
+        # FIXME: trim nextoffsets
+        return out2, nextoffsets
```

Comment:
Reminder of this FIXME: it looks important and easily overlooked.

Reply:
the function is not used, I removed it